src/openingHours.ts

```diff
@@ -104,6 +115,150 @@ export class OpeningHours {
     return this.isOpenOn(new Date());
   }
 
+  /**
+   * Returns next day open
+   */
+  private getNextOpeningDay(date: Date): NextOpeningDay {
+    const today = date.getDay();
+    let times;
+    let i = today;
+
+    for (let key in this.openingHours) {
+      i = i === 7 ? 0 : i + 1;
+      times = this.getTimesOfDay(i);
+      if (times.length > 0) {
+        times = this.openingHours[key];
+        break;
+      }
+    }
+    return {
+      times,
+      nextDay: SIMPLE_DAYS_NAMES[i],
+      isTomorrow: i === today + 1 ? true : false,
+    };
+  }
+
+  private openAt = (formatHours, date: Date, nextDay): string | null => {
+    if (formatHours.length > 1) {
+      const now = this.constructDateFromTime(`${date.getHours()}:${date.getMinutes()}`);
+      if (nextDay) {
+        return formatHours[0].hour;
+      }
+      for (let i = 0; i < formatHours.length - 1; i = i + 2) {
+        if (now < formatHours[i].date) {
+          console.log(i);
+          return formatHours[i].hour;
+        }
+      }
+    }
+    return null;
+  };
+
+  private openUntil = (formatHours: FormatHours[], date: Date): string => {
+    if (formatHours.length > 1) {
+      const now = this.constructDateFromTime(`${date.getHours()}:${date.getMinutes()}`);
+      for (let i = 1; i < formatHours.length; i = i + 2) {
+        if (now < formatHours[i].date || now > formatHours[formatHours.length - 1].date) {
+          return formatHours[i].hour;
+        }
+      }
+    }
+    return `Aujourd'hui`;
+  };
+
+  public checkOpeningTime = (date: Date) => {
+    const isOpenNow = this.isOpenOn(date);
+
+    const allHours = this.getTable();
+
+    const dayIndex = date.getDay();
+    const daySimpleIndex = SIMPLE_DAYS_NAMES[dayIndex];
+    const todayHours = allHours[daySimpleIndex];
+    if (!todayHours || todayHours.length === 0) {
+      return this.nextDayOpeningTime(allHours, date);
+    }
+    const formatHours: FormatHours[] = [];
+    todayHours.map(todayHour =>
+      todayHour
+        .split('-')
+        .map(hour => formatHours.push({ date: this.constructDateFromTime(hour), hour })),
+    );
+    if (isOpenNow) {
+      return {
+        open: true,
+        openUntil: this.openUntil(formatHours, date),
+        openAt: null,
+        weekDay: null,
+      };
+    }
+    if (!isOpenNow) {
+      const hour = this.openAt(formatHours, date, undefined);
+
+      if (!hour) {
+        return this.nextDayOpeningTime(allHours, date);
+      }
+      return {
+        open: false,
+        openUntil: null,
+        openAt: hour,
+        weekDay: null,
```

Comment:
Okay I thought of how we can do it more easily understandable.

WHat do you think of having 
```
{
        open: false,
        openUntil: null,
        reopensAt: hour,
        reopensInDay: number, // 0 if today, 1 if tomorrow, 2 if it reopens it two days... 
}
```

That way we know that `openUntil` exists only when `open` is `true`. And `reopensAt` means exactly "when it will re-open", and `reopensInDay`: give the day corresponding to `resopensAt`.